native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/AgentCommandLineProvider.java

```diff
@@ -79,6 +79,9 @@ public AgentCommandLineProvider() {
         if (getEnabled().get()) {
             File outputDir = getOutputDirectory().getAsFile().get();
             List<String> agentOptions = new ArrayList<>(getAgentOptions().getOrElse(Collections.emptyList()));
+            if (agentOptions.contains("config-output-dir")) {
```

Comment:
That check won't suffice. 

You'll need to iterate (or stream) over the supplied options and check if any of the individual strings contains that, since a violation would be in the form of `"config-out-dir=<user supplied output dir>"`.

Reply:
Of course you're right :facepalm: Fixed!